tracking-data-streaming/src/cfn_template/iotResources.yml

```diff
@@ -0,0 +1,114 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: A Template to provision AWS IoT Core Resources for the Amazon Location Live Device Tracking and Geofencing sample app.
+Parameters:
+  TrackerName:
+    Type: String
+    Description: Name of the Amazon Location Tracker to use.
+Resources:
+  TrackingAndGeofencingSampleIoTLambdaFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - lambda.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
+      - PolicyName: TrackingAndGeofencingSampleIoTLambdaFunctionPolicy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - geo:BatchUpdateDevicePosition
+            Resource: !Sub 
+              - 'arn:aws:geo:${AWS::Region}:${AWS::AccountId}:tracker/${TrackerName}'
+              - TrackerName: !Ref TrackerName
+
+  TrackingAndGeofencingSampleIoTLambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python3.12
+      FunctionName: TrackingAndGeofencingSampleIoTLambdaFunction
+      Role: !GetAtt TrackingAndGeofencingSampleIoTLambdaFunctionRole.Arn
+      Environment:
+        Variables:
+          TrackerName: !Ref TrackerName
+      Handler: index.lambda_handler
+      Code:
+        ZipFile: |
+          from datetime import datetime
+          import json
+          import os
+
+          import boto3
+
+          # Update this to match the name of your Tracker resource
+          TRACKER_NAME = os.environ['TrackerName']
+
+          """
+          This Lambda function receives a payload from AWS IoT Core and publishes device updates to 
```

Comment:
The purpose of this sample is to publish position updates to Kinesis Data Stream and display the updates on a map. This Lambda function seems to be taking another direction, so I am not sure about adding it. It can cause confusion for the customer.

Also, what is calling this lambda?

I would image we have a separate but similar sample to demo the IoT -> Lambda -> Location route.

Reply:
I disagree that it adds confusion, as it just adds another path for data to take. The sample will show data no matter if it comes from Kinesis or AWS IoT Core. A tracker is a tracker and should be shown on the map. This lambda is being called from the IoT Rule and is invoked everytime a message comes in via IoT Core on that location MQTT topic.

Reply:
There will be a disconnection about the user experiences between using Kinesis and using IoT if this PR is merged at the current state. 

The current demo is that when the "Run Demo" button is clicked, the web app sends the position updates to Kinesis, and they get streamed to the tracker and displayed on the map.

With this change, we allow the intake from IoT Core, but nothing is calling IoT Core to pass the position updates. To keep the user experience consistent, I imagine there will be "Run Demo with IoT" and "Run Demo with Kinesis" buttons instead, where the "Run Demo with IoT" button will trigger calls from the web app to IoT.

Also this lambda function should be a separate deployable (e.g. an SAR) like the Kinesis one.

I love the change to add analytics to this sample. I just think if we add IoT to the sample, we need to make the customer experience consistent for using either Kinesis or IoT as the source.